src/parse.ts

```diff
@@ -12,12 +12,18 @@ import { Grammar, Parser } from "nearley";
 
 import { buildRos2Type } from "./buildRos2Type";
 import ros1Rules from "./ros1.ne";
+import ros2idlRules from "./ros2idl.ne";
 
 const ROS1_GRAMMAR = Grammar.fromCompiled(ros1Rules);
+const ROS2IDL_GRAMMAR = Grammar.fromCompiled(ros2idlRules);
 
 export type ParseOptions = {
-  /** Parse message definitions as ROS2. Otherwise, parse as ROS1 */
+  /** Parse message definitions as ROS 2. Otherwise, parse as ROS1 */
   ros2?: boolean;
+  /** Parse message definitions in the ROS 2 subset of IDL.
+   * Ignores ros2 and skipTypeFixup option value.
+   */
+  ros2idl?: boolean;
```

Comment:
maybe time to change this to `syntax?: "ros1" | "ros2msg" | "ros2idl"` (open to bikeshedding on names)?

Reply:
On the other hand, I am left wondering why these are all in one `parse()` function. Maybe we should just have `parseRos1Msg`, `parseRos2Msg`, and `parseRos2Idl` functions.

Reply:
I too wonder this. In my mind all the downstream places where we parse a scheme we have an encoding that would tell us which type of parser to use.

Reply:
I'm more partial to the enums or separate functions, but this is backwards compatible. Don't know what our priorities around this are though

Reply:
updated to have a separate `parseRos2idl` function in `parse.ts`, allowing for only a minor version change